### PR TITLE
update with multitargeting and deps.json info

### PR DESCRIPTION
**Pull request by ghogen**
## Summary

Describe your changes here.

Fixes #Issue_Number, dotnet/docs#Issue_Number or dotnet/dotnet-api-docs#Issue_Number (if available)

### CODE REVIEW

File: msbuild/custom-task-code-generation/README.md [hunk 1]
```diff
@@ -122,6 +122,40 @@ We are going to generate a NuGet package, so first we need to add some basic inf
...
+   <PackageReference
+      Include="Microsoft.Build.Utilities.Core"
+      Version="17.0.0" />
+   <PackageReference
+      Include="Microsoft.Extensions.DependencyInjection"
+      Version="6.0.0" />
+</ItemGroup>
+```
+
+Now, mark every dependency of this Task project, both `PackageReference` and `ProjectReference` with the `PrivateAssets="all"` attribute. This will tell NuGet not to expose these dependencies to consuming projects at all. You can read more about controlling dependency assets [at the NuGet documentation](https://docs.microsoft.com/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets).
```
Comment (gewarren) [inline msbuild/custom-task-code-generation/README.md:140]:
```suggestion
Now, mark every dependency of this Task project, both `PackageReference` and `ProjectReference`, with the `PrivateAssets="all"` attribute. This tells NuGet not to expose these dependencies to consuming projects at all. You can read more about controlling dependency assets [in the NuGet documentation](https://docs.microsoft.com/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets).
```

**Review by gewarren — APPROVED** [review]

**Comment by ghogen** [comment]
@baronfel 
Can you take a look at these changes - this ports changes you made a while back in our tutorial into the sample code and README

File: msbuild/custom-task-code-generation/AppSettingStronglyTyped/AppSettingStronglyTyped/AppSettingStronglyTyped.csproj [hunk 2]
```diff
@@ -9,14 +9,21 @@
...
     <ItemGroup>
-        <PackageReference Include="Microsoft.Build.Utilities.Core" Version="17.0.0" />
+	   <PackageReference 
+		   Include="Microsoft.Build.Utilities.Core"
+		   Version="17.0.0"
+           PrivateAssets="all" />
+	   <PackageReference
+		   Include="Microsoft.Extensions.DependencyInjection"
+		   Version="6.0.0"
+           PrivateAssets="all" />
```
Comment (baronfel) [inline msbuild/custom-task-code-generation/AppSettingStronglyTyped/AppSettingStronglyTyped/AppSettingStronglyTyped.csproj:26]:
```suggestion
	   <PackageReference 
		       Include="Microsoft.Build.Utilities.Core"
    		   Version="17.0.0"
           PrivateAssets="all" />
	   <PackageReference
		       Include="Microsoft.Extensions.DependencyInjection"
    		   Version="6.0.0"
           PrivateAssets="all" />
```

nit: alignment :)

File: msbuild/custom-task-code-generation/README.md [hunk 3]
```diff
@@ -131,29 +165,57 @@ Then, the dependencies of your MSBuild task must be packaged inside the package,
...
+        <Content Include="build\AppSettingStronglyTyped.props" PackagePath="build\" />
+        <Content Include="build\AppSettingStronglyTyped.targets" PackagePath="build\" />
     </ItemGroup>
 
     <Target Name="CopyProjectReferencesToPackage" DependsOnTargets="ResolveReferences">
         <ItemGroup>
-            <!-- the dependencies of your MSBuild task must be packaged inside the package, they cannot be expressed as normal PackageReferences -->
+            <!-- The dependencies of your MSBuild task must be packaged inside the package, they cannot be expressed as normal PackageReferences -->
 
             <!--example: <BuildOutputInPackage Include="$(PkgFParsec)/lib/netstandard2.0/FParsecCS.dll" />-->
```
Comment (baronfel) [inline msbuild/custom-task-code-generation/README.md:202]: in this README, as well as in the csproj for the app-settings task, the `CopyProjectReferencesToPackage` target is missing the line that actually copies the PackageReference's runtime assets into the package. That line is:

```xml
<BuildOutputInPackage
      Include="@(ReferenceCopyLocalPaths)"
      TargetPath="%(ReferenceCopyLocalPaths.DestinationSubPath)" />
```

**Review by baronfel — APPROVED** [review]
One note about the examples that should be addressed, but otherwise :shipit:

**Comment by ghogen** [comment]
#sign-off

**Pull request merged**